src/main/java/com/pinterest/secor/io/impl/FlexibleDelimitedFileReaderWriterFactory.java

```diff
@@ -0,0 +1,166 @@
+      codec.createInputStream(inputStream,
+      mDecompressor = CodecPool.getDecompressor(codec)));
+      this.mOffset = path.getOffset();
+    }
+
+    public byte delimterReader() {
+      byte delimiter = '\n';
+      try {
+        String readerDelimiter = SecorConfig.load().getFileReaderDelimiter();
+        if (!readerDelimiter.isEmpty()){
+          delimiter = (byte)readerDelimiter.charAt(0);
+        }
+      }catch(ConfigurationException e){}
+        return delimiter;
+      }
+
+      @Override
+      public KeyValue next() throws IOException {
+        ByteArrayOutputStream messageBuffer = new ByteArrayOutputStream();
+        int nextByte;
+        while ((nextByte = mReader.read()) != DELIMITER) {
+          if (nextByte == -1) { // end of stream?
+            if (messageBuffer.size() == 0) { // if no byte read
+              return null;
+            } else { // if bytes followed by end of stream: framing error
+              throw new EOFException(
+              "Non-empty message without delimiter");
+            }
+          }
+          messageBuffer.write(nextByte);
+        }
+        return new KeyValue(this.mOffset++, messageBuffer.toByteArray());
+      }
+
+      @Override
+      public void close() throws IOException {
+        this.mReader.close();
+        CodecPool.returnDecompressor(mDecompressor);
+        mDecompressor = null;
+      }
+    }
+
+    protected class FlexibleDelimitedFileWriter implements FileWriter {
+      private final CountingOutputStream mCountingStream;
+      private final BufferedOutputStream mWriter;
+      private Compressor mCompressor = null;
+      private byte DELIMITER = delimterWriter();
+
+      public FlexibleDelimitedFileWriter(LogFilePath path, CompressionCodec codec) throws IOException {
+        Path fsPath = new Path(path.getLogFilePath());
+        FileSystem fs = FileUtil.getFileSystem(path.getLogFilePath());
+        this.mCountingStream = new CountingOutputStream(fs.create(fsPath));
+        this.mWriter = (codec == null) ? new BufferedOutputStream(
+        this.mCountingStream) : new BufferedOutputStream(
+        codec.createOutputStream(this.mCountingStream,
+        mCompressor = CodecPool.getCompressor(codec)));
+      }
+
+      public byte delimterWriter() {
+        byte delimiter = '\\';
+        try {
+          String writerDelimiter = SecorConfig.load().getFileWriterDelimiter();
+          if (!writerDelimiter.isEmpty()){
+            delimiter = (byte)writerDelimiter.charAt(0);
+          }
+        }catch(ConfigurationException e){}
+          return delimiter;
+        }
+
+        @Override
+        public long getLength() throws IOException {
+          assert this.mCountingStream != null;
+          return this.mCountingStream.getCount();
+        }
+
+        @Override
+        public void write(KeyValue keyValue) throws IOException {
+          this.mWriter.write(keyValue.getValue());
+          if (DELIMITER != (byte)'\\'){
```

Comment:
`byte` value can't be for an empty `String` , and `\` is an escape character in creation of a string  so it can't be used on its own to be added between messages pulled out of secor as an   single message of `\` which would mean empty message with just `\`